noaa/gefs/forecast/reformat.py

```diff
@@ -248,29 +271,79 @@ def reformat_kubernetes(
 
     dataset_id = template_ds.attrs["dataset_id"]
     job_name = f"{dataset_id}-{job_timestamp}"
-    kubernetes_job = string_template.substitute(
-        "deploy/kubernetes_ingest_job.yaml",
-        {
-            "NAME": job_name,
-            "IMAGE": image_tag,
-            "DATASET_ID": dataset_id,
-            "INIT_TIME_END": pd.Timestamp(init_time_end).isoformat(),
-            "WORKERS_TOTAL": workers_total,
-            "PARALLELISM": parallelism,
-            "CPU": 6,
-            "MEMORY": "40G",
-            "EPHEMERAL_STORAGE": "60G",
-        },
+
+    kubernetes_job = ReformatJob(
+        image=image_tag,
+        dataset_id=dataset_id,
+        workers_total=workers_total,
+        parallelism=parallelism,
+        cpu="14",  # fits on 16 core nodes
+        memory="58G",  # fits in 64GB nodes
+        ephemeral_storage="150G",
+        command=[
+            "reformat-chunks",
+            pd.Timestamp(init_time_end).isoformat(),
+        ],
+    )
+    kubernetes.config.load_kube_config()
+    batch_v1 = kubernetes.client.BatchV1Api()
+    batch_v1.create_namespaced_job(
+        namespace="default",
+        body=kubernetes.client.V1Job(**kubernetes_job.get_kubernetes_job()),
     )
 
-    subprocess.run(  # noqa: S603
-        ["/usr/bin/kubectl", "apply", "-f", "-"],
-        input=kubernetes_job,
-        text=True,
+    logger.info(f"Submitted kubernetes job {job_name}")
+
+
+def deploy_operational_updates() -> None:
+    job_timestamp = pd.Timestamp.now("UTC").strftime("%Y-%m-%dt%M-%S")
+
+    docker_repo = os.environ["DOCKER_REPOSITORY"]
+    assert re.fullmatch(r"[0-9a-zA-Z_\.\-\/]{1,1000}", docker_repo)
+    image_tag = f"{docker_repo}:{job_timestamp}"
+
+    subprocess.run(  # noqa: S603  allow passing variable to subprocess, it's realtively sanitized above
+        [
+            "/usr/bin/docker",
+            "build",
+            "--file",
+            "deploy/Dockerfile",
+            "--tag",
+            image_tag,
+            ".",
+        ],
         check=True,
     )
+    subprocess.run(  # noqa: S603
+        ["/usr/bin/docker", "push", image_tag], check=True
+    )
+    logger.info(f"Pushed {image_tag}")
 
-    logger.info(f"Submitted kubernetes job {job_name}")
+    workers_total = 20
+    parallelism = min(workers_total, 10)
```

Comment:
```suggestion
    workers_total = 1
    parallelism = 1
```
For correct updates without fancy synchronization we need to ensure only one process. Actually, that reminds me, will you set a max concurrent jobs to 1 on the cronjob?

Reply:
Good call, fixed the workers & parallelism and set the `CronJob`'s `concurrencyPolicy` to `Replace`